### PR TITLE
Upgrading SQLModel to the latest version

### PR DESCRIPTION
**Pull request by bcdurak**
## Describe changes
I upgraded the version of the [SQLModel](https://sqlmodel.tiangolo.com) package that we are using from `0.0.8` to `0.0.16`.

The most critical factor in this upgrade stems from [this PR right here](https://github.com/tiangolo/sqlmodel/pull/669). With this change, they have changed the way that they handle `Enum` values.

For instance, if you are familiar with our [component schema](https://github.com/zenml-io/zenml/blob/main/src/zenml/zen_stores/schemas/component_schemas.py) (which we defined through SQLModel), we added a field called `type` which was a `StackComponentType`:

```python
# the schema of the stack component
class StackComponentSchema(...):
    ...
    type: StackComponentType
    ...
    
# and the stack component type looked like this:
class StackComponentType(StrEnum):
    ...
    ARTIFACT_STORE = "artifact_store"
    ...
```

With this setup, when we registered, for instance, a new artifact store, we created an entry in the components table of our DB where the column `type` had the string `artifact_store` stored in it as a value. However, with the new changes, SQLModel now gives higher priority to `Enum` fields and saves the value `ARTIFACT_STORE` instead. While this is alright if you are starting from scratch, if you have any entry in a table with an `Enum` field zenml will fail after the upgrade. Instead of taking the migration route, we decided to adjust our schemas to use str fields instead and updated the corresponding `to_model`, `update` and `from_request` methods.

@avishniakov every enum field is using a similar approach to the one that you took with the `TagSchema`s and their color field.

**Additionally**, I have removed `fastapi_utils` as a dependency as it was completely unused and tweaked the code a bit to work with the new SQLModel version.

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [X] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [X] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by strickvl** [comment]
@coderabbitai review

**Review by avishniakov — CHANGES_REQUESTED** [review]
Agree with most things @coderabbitai highlighted + test are failing now

File: src/zenml/integrations/evidently/__init__.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ class EvidentlyIntegration(Integration):
     """[Evidently](https://github.com/evidentlyai/evidently) integration for ZenML."""
 
     NAME = EVIDENTLY
-    REQUIREMENTS = ["evidently>0.2.6,<0.4.5"]  # supports pyyaml 6
+    REQUIREMENTS = ["evidently>0.2.6,<0.4.5", "sqlmodel<0.0.10"]  # supports pyyaml 6
```
Comment (strickvl) [inline src/zenml/integrations/evidently/__init__.py:57]: should add a comment on why we've added sqlmodel here

File: src/zenml/integrations/airflow/__init__.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ class AirflowIntegration(Integration):
     """Definition of Airflow Integration for ZenML."""
 
     NAME = AIRFLOW
-    REQUIREMENTS = ["apache-airflow~=2.4.0"]
+    REQUIREMENTS = ["apache-airflow~=2.4.0", "sqlmodel<0.0.10"]
```
Comment (strickvl) [inline src/zenml/integrations/airflow/__init__.py:33]: should add a comment on why we've added sqlmodel here

File: src/zenml/integrations/evidently/__init__.py (hunk 1 above)
Comment (strickvl) [reply src/zenml/integrations/evidently/__init__.py:57]: and maybe a note in the docs that this is what's happening?

File: src/zenml/integrations/airflow/__init__.py (hunk 2 above)
Comment (strickvl) [reply src/zenml/integrations/airflow/__init__.py:33]: and maybe a note in the docs that this is what's happening?

File: src/zenml/integrations/great_expectations/__init__.py [hunk 3]
```diff
@@ -35,6 +35,7 @@ class GreatExpectationsIntegration(Integration):
         "great-expectations>=0.15.0,<=0.15.47",
         # typing_extensions 4.6.0 and above doesn't work with GE
         "typing_extensions<4.6.0",
+        "sqlmodel<0.0.10",
```
Comment (strickvl) [inline src/zenml/integrations/great_expectations/__init__.py:38]: should add a comment on why we've added sqlmodel here. and maybe a note in the docs that this is what's happening?

**Review by strickvl — CHANGES_REQUESTED** [review]
LGTM aside from the comments I already made.

**Review by strickvl — APPROVED** [review]

File: src/zenml/integrations/airflow/__init__.py [hunk 4]
```diff
@@ -32,7 +32,14 @@ class AirflowIntegration(Integration):
...
     # remove pendulum version requirement once Airflow supports
     # pendulum>-3.0.0
-    REQUIREMENTS = ["apache-airflow~=2.4.0", "pendulum<3.0.0"]
+    REQUIREMENTS = [
+        "apache-airflow~=2.4.0",
+        "pendulum<3.0.0",
+        # We need to add this as an extra dependency to manually downgrade
+        # SQLModel. Otherwise, the initial installation of ZenML installs
+        # a higher version SQLModel and a version mismatch is created.
+        "sqlmodel>=0.0.9,<=0.0.16",
```
Comment (avishniakov) [inline src/zenml/integrations/airflow/__init__.py:41]: Is this still needed? You have exactly the same in `pyproject.toml` now...

File: src/zenml/integrations/evidently/__init__.py [hunk 5]
```diff
@@ -54,7 +54,13 @@ class EvidentlyIntegration(Integration):
     """[Evidently](https://github.com/evidentlyai/evidently) integration for ZenML."""
 
     NAME = EVIDENTLY
-    REQUIREMENTS = ["evidently>0.2.6,<0.4.5"]  # supports pyyaml 6
+    REQUIREMENTS = [
+        "evidently>0.2.6,<0.4.5",  # supports pyyaml 6
+        # We need to add this as an extra dependency to manually downgrade
+        # SQLModel. Otherwise, the initial installation of ZenML installs
+        # a higher version SQLModel and a version mismatch is created.
+        "sqlmodel>=0.0.9,<=0.0.16"
```
Comment (avishniakov) [inline src/zenml/integrations/evidently/__init__.py:62]: same

File: src/zenml/integrations/airflow/__init__.py (hunk 4 above)
Comment (bcdurak) [reply src/zenml/integrations/airflow/__init__.py:41]: Yes, if we don't put this here, when someone installs ZenML, it installs sqlmodel 0.0.16. When they install the airflow integration on top, it downgrades sqlalchemy but not the sqlmodel, which creates a drift.

**Comment by david101-hunter** [comment]
Hi, after I upgrade sqlmodel version, I encounter issue
`zenml 0.56.3 requires sqlmodel==0.0.8, but you have sqlmodel 0.0.16 which is incompatible.`

I want to intergate feast (0.36.0), however:

`zenml 0.56.3 requires pydantic<1.11,>=1.9.0`
`feast 0.36.0 requires pydantic>=2.0.0, but you have pydantic 1.10.15 which is incompatible`

How I can fix that.

**Pull request merged**